src/ctlib/unittests/row_count.c

```diff
@@ -122,11 +122,16 @@ test(int final_rows, int no_rows)
 	CS_RETCODE ret;
 	CS_CHAR cmdbuf[4096];
 	char results[1024];
+        char *results_end = results, *cmdbuf_end = cmdbuf;
 
 	run_command(cmd, "create table #tmp1 (i int not null)");
 
-	results[0] = 0;
-	strcpy(cmdbuf, "create proc sample_rpc as ");
+#define strcat(s, k) do { \
```

Comment:
pretty ugly (maybe more ugly the bug). Why not using strlcpy/strlcat (include replacements.h for portability).

Reply:
This approach takes advantage of known lengths to be marginally faster, but I suppose that's a matter of premature optimization, given that this program is (a) a test and (b) almost certainly I/O bound.  I presume you'd at that point prefer to do away with this `#define` altogether and just call `strlcpy`/`strlcat` directly?